### PR TITLE
Remove CommitID dependency to update a device

### PR DESCRIPTION
**Pull request by Saurabh-Thakre**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/devices.go [hunk 1]
```diff
@@ -461,6 +462,36 @@ func (s *DeviceService) SetDeviceUpdateAvailability(account string, deviceID uin
...
+		return 0, result.Error
+	}
+	// check for updates , find if any later images exists to get the commitID
+	var updateImages []models.Image
+	if result := db.DB.Where("account = ? AND image_set_id = ? AND status = ? AND created_at > ?", deviceImage.Account, deviceImage.ImageSetID, models.ImageStatusSuccess, deviceImage.CreatedAt).Order("version desc").Find(&updateImages); result.Error != nil {
+		return 0, result.Error
+	}
+
+	if len(updateImages) == 0 {
+		return 0, new(DeviceHasNoImage)
```
Comment (ldjebran) [inline pkg/services/devices.go:488]: may be create a new error:  DeviceHasNoImageUpdate   (as the device has an image at this stage)

File: pkg/services/devices.go [hunk 2]
```diff
@@ -461,6 +462,36 @@ func (s *DeviceService) SetDeviceUpdateAvailability(account string, deviceID uin
 	return nil
 }
 
+// GetUpdateCommitFromDevice fetches the commitID from the latest Device Image
+func (s *DeviceService) GetUpdateCommitFromDevice(account string, deviceID string) (uint, error) {
```
Comment (ldjebran) [inline pkg/services/devices.go:466]: Please create a unitest for this function

**Review by ldjebran — COMMENTED** [review]
Looks good , left some comments

**Comment by AaronH88** [comment]
Looking good.
As Djebran said, once we have a few tests it will be good to merge :smile:

File: pkg/services/devices_test.go [hunk 3]
```diff
@@ -689,4 +689,37 @@ var _ = Describe("DeviceService", func() {
 			Expect(savedDeviceGroup.Devices).To(BeEmpty())
 		})
 	})
+	Context("Get CommitID from Device Image", func() {
+		When("device Image doest not exists", func() {
```
Comment (ldjebran) [inline pkg/services/devices_test.go:693]: do not see calling function GetUpdateCommitFromDevice

File: pkg/services/devices_test.go [hunk 4]
```diff
@@ -689,4 +689,37 @@ var _ = Describe("DeviceService", func() {
...
+				device := models.Device{
+					Account: account,
+					UUID:    faker.UUIDHyphenated(),
+				}
+				result := db.DB.Where(models.Device{Account: device.Account, UUID: device.UUID}).First(&device)
+				Expect(result.Error).To(MatchError("record not found"))
+				Expect(device.ImageID == 0).To(BeTrue())
+			})
+		})
+		When("device Image does not have update", func() {
```
Comment (ldjebran) [inline pkg/services/devices_test.go:705]: same here: do not see calling function GetUpdateCommitFromDevice

**Comment by Saurabh-Thakre** [comment]
Closed this one and opened a new PR #917 917

**Pull request closed without merging**